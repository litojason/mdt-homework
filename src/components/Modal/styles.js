import styled from "styled-components";

import { borderRadius, colors, fonts, shadow, spacing } from "constants/styles";

export const Background = styled.div`
  position: fixed;
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100vw;
  height: 100vh;
  padding: ${spacing.base};
  background-color: ${colors.modal};
  box-sizing: border-box;
  z-index: 20;
`;

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  padding: ${spacing.base};
  border-radius: ${borderRadius.base};
  background-color: ${colors.cardBg};
  box-shadow: ${shadow.card};
`;

export const Title = styled.div`
  font-size: ${fonts.size.title};
  font-weight: 600;
`;

export const Description = styled.div`
  margin: ${spacing.half} 0 ${spacing.base} 0;
  text-align: center;
`;
