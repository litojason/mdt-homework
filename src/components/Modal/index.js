import React from "react";
import PropTypes from "prop-types";

import { Background, Container, Description, Title } from "./styles";
import Button from "components/Button";

const Modal = ({
  title,
  description,
  setModalVisible,
  cancelable,
  buttonText,
  buttonPress = null,
}) => {
  return (
    <Background onClick={() => (cancelable ? setModalVisible(false) : null)}>
      <Container>
        <Title>{title}</Title>

        <Description>{description}</Description>

        <Button
          onClick={() => {
            setModalVisible(false);
            if (buttonPress) buttonPress();
          }}
        >
          {buttonText}
        </Button>
      </Container>
    </Background>
  );
};

Modal.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
  setModalVisible: PropTypes.func,
  cancelable: PropTypes.bool,
  buttonText: PropTypes.string,
  buttonPress: PropTypes.oneOfType([PropTypes.func, PropTypes.string]),
};

export default Modal;
