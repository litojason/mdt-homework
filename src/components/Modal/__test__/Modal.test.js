import React from "react";
import renderer from "react-test-renderer";
// import { shallow, mount } from "enzyme";
// import "jest-enzyme";
import "jest-styled-components";

import Modal from "../index";

describe("Component: Modal", () => {
  it("should render when modal is true", () => {
    const component = <Modal />;

    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
