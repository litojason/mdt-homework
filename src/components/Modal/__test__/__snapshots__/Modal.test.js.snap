// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Modal should render 1`] = `
.c0 {
  position: fixed;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100vw;
  height: 100vh;
  padding: 1rem;
  background-color: rgba(0,0,0,0.5);
  box-sizing: border-box;
  z-index: 20;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  padding: 1rem;
  border-radius: 0.5rem;
  background-color: #FFF;
  box-shadow: 0px 3px 6px 0px rgba(0,0,0,0.125);
}

.c2 {
  font-size: 1.5rem;
  font-weight: 600;
}

.c3 {
  margin: 0.5rem 0 1rem 0;
  text-align: center;
}

.c4 {
  padding: 1rem;
  background-color: #E72934;
  border: 0;
  color: #FFF;
  border-radius: 0.5rem;
  opacity: 1;
}

<div
  className="c0"
  onClick={[Function]}
>
  <div
    className="c1"
  >
    <div
      className="c2"
    />
    <div
      className="c3"
    />
    <button
      className="c4"
      loading="false"
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`Component: Modal should render when modal is true 1`] = `
.c0 {
  position: fixed;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100vw;
  height: 100vh;
  padding: 1rem;
  background-color: rgba(0,0,0,0.5);
  box-sizing: border-box;
  z-index: 20;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  padding: 1rem;
  border-radius: 0.5rem;
  background-color: #FFF;
  box-shadow: 0px 3px 6px 0px rgba(0,0,0,0.125);
}

.c2 {
  font-size: 1.5rem;
  font-weight: 600;
}

.c3 {
  margin: 0.5rem 0 1rem 0;
  text-align: center;
}

.c4 {
  padding: 1rem;
  background-color: #E72934;
  border: 0;
  color: #FFF;
  border-radius: 0.5rem;
  opacity: 1;
}

<div
  className="c0"
  onClick={[Function]}
>
  <div
    className="c1"
  >
    <div
      className="c2"
    />
    <div
      className="c3"
    />
    <button
      className="c4"
      loading="false"
      onClick={[Function]}
    />
  </div>
</div>
`;
